rust/private/rustc.bzl

```
@@ -905,8 +905,11 @@ def construct_arguments(
     rustc_flags.add_all(rust_std_paths, before_each = "-L", format_each = "%s")
     rustc_flags.add_all(rust_flags)
 
+    # Gather data path from crate_info since it is inherited from real crate for rust_doc and rust_test
+    data_paths = depset(direct = getattr(attr, "data", []), transitive = [crate_info.compile_data_targets])
+
     # Deduplicate data paths due to https://github.com/bazelbuild/bazel/issues/14681
```

It's the `depset` construction which does the deduplication, so let's leave the comment on that piece

```suggestion
    # Gather data path from crate_info since it is inherited from real crate for rust_doc and rust_test
    # Deduplicate data paths due to https://github.com/bazelbuild/bazel/issues/14681
    data_paths = depset(direct = getattr(attr, "data", []), transitive = [crate_info.compile_data_targets])
```